src/Microsoft.AspNetCore.DataProtection.Extensions/DataProtectionProvider.cs

```diff
@@ -150,7 +150,7 @@ public static IDataProtectionProvider Create(string applicationName, X509Certifi
             return CreateProvider(keyDirectory, setupAction, certificate);
         }
 
-        private static IDataProtectionProvider CreateProvider(
+        internal static IDataProtectionProvider CreateProvider(
```

Comment:
Pinging @Eilon to get his opinion on private to internal change here.

Reply:
Eh, this is not a test-only change. I'd rather not take this through.

Reply:
I'd rather even do an `#ifdef` or something to avoid any possible change to the DLL (even if we don't ship the DLL).

Reply:
Can someone explain the risk you're worried about? I'm having a hard time thinking of a realistic way that this would break someone.